edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/versioning/ModuleManagementHttpClientVersioned.cs

```diff
@@ -23,14 +24,18 @@ abstract class ModuleManagementHttpClientVersioned
         const string LogsUrlTailParameter = "tail";
         const string LogsUrlSinceParameter = "since";
 
+        const int MaxRetryCount = 3;
+
         static readonly TimeSpan DefaultOperationTimeout = TimeSpan.FromMinutes(5);
 
         static readonly RetryStrategy TransientRetryStrategy =
-            new ExponentialBackoff(retryCount: 3, minBackoff: TimeSpan.FromSeconds(2), maxBackoff: TimeSpan.FromSeconds(30), deltaBackoff: TimeSpan.FromSeconds(3));
+            new ExponentialBackoff(retryCount: MaxRetryCount, minBackoff: TimeSpan.FromSeconds(2), maxBackoff: TimeSpan.FromSeconds(30), deltaBackoff: TimeSpan.FromSeconds(3));
 
         readonly ITransientErrorDetectionStrategy transientErrorDetectionStrategy;
         readonly TimeSpan operationTimeout;
 
+        int failedSocketCount = 0;
```

Comment:
why is this needed? I tough it can not recover from this exception that's why it needs to restart the module. If needed it should be atomic because methods from this class can be called from different threads. There is an AtomicLong implementation in Utils that you can use.

Reply:
Sure I can make it failed right away. 
I wasn't sure if there is other type of Socket failure that would cause the "connection refused' that may recover.